Nightly GitHub Actions Build Fail on Thu, December 8th 2022

See run for more details:
https://github.com/TileDB-Inc/TileDB/actions/runs/3644826733

Not sure why CI wouldn't have caught this before merge....
```
[  5%] Building CXX object tiledb/common/thread_pool/CMakeFiles/thread_pool.dir/thread_pool.cc.o
In file included from /Users/runner/work/TileDB/TileDB/tiledb/platform/cert_file.cc:33:
/Users/runner/work/TileDB/TileDB/tiledb/../tiledb/platform/cert_file.h:133:30: error: 'exists' is unavailable: introduced in macOS 10.15
        if (std::filesystem::exists(cert, ec)) {
                             ^
/Applications/Xcode_13.2.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.1.sdk/usr/include/c++/v1/filesystem:1906:39: note: 'exists' has been explicitly marked unavailable here
inline _LIBCPP_INLINE_VISIBILITY bool exists(const path& __p,
                                      ^
In file included from /Users/runner/work/TileDB/TileDB/tiledb/platform/cert_file.cc:33:
/Users/runner/work/TileDB/TileDB/tiledb/../tiledb/platform/cert_file.h:133:37: error: 'path' is unavailable: introduced in macOS 10.15
        if (std::filesystem::exists(cert, ec)) {
                                    ^
/Applications/Xcode_13.2.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.1.sdk/usr/include/c++/v1/filesystem:941:3: note: 'path' has been explicitly marked unavailable here
  path(const _Source& __src, format = format::auto_format) {
  ^
In file included from /Users/runner/work/TileDB/TileDB/tiledb/platform/cert_file.cc:33:
/Users/runner/work/TileDB/TileDB/tiledb/../tiledb/platform/cert_file.h:133:37: error: '~path' is unavailable: introduced in macOS 10.15
        if (std::filesystem::exists(cert, ec)) {
                                    ^
/Applications/Xcode_13.2.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX12.1.sdk/usr/include/c++/v1/filesystem:961:3: note: '~path' has been explicitly marked unavailable here
  ~path() = default;
  ^
3 errors generated.
make[5]: *** [tiledb/CMakeFiles/TILEDB_CORE_OBJECTS.dir/platform/cert_file.cc.o] Error 1
make[4]: *** [tiledb/CMakeFiles/TILEDB_CORE_OBJECTS.dir/all] Error 2
make[4]: *** Waiting for unfinished jobs....
[  5%] Built target thread_pool
```

--- a/tiledb/platform/cert_file.h
+++ b/tiledb/platform/cert_file.h
@@ -34,13 +34,28 @@
 #define TILEDB_CERT_FILE_H
 
 #include <array>
-#include <filesystem>
 #include <mutex>
 #include <string>
 #include <system_error>
 
 #include "tiledb/common/logger.h"
 #include "tiledb/platform/os.h"
+
+// We can remove this preprocessor block once our minimum
+// macOS version is at least 10.15. Until then macOS won't
+// compile even though this std::filesystem is only run
+// on Linux.
+#if defined(__APPLE__) && defined(__MACH__)
+static bool file_exists(const std::string&) {
+  throw std::runtime_error("cert file discovery not supported");
+}
+#else
+#include <filesystem>
+static bool file_exists(const std::string& path) {
+  std::error_code ec;
+  return std::filesystem::exists(path, ec);
+}
+#endif
 
 namespace tiledb::platform {
 
@@ -129,8 +144,7 @@ class CertFileTraits<LinuxOS> {
       };
 
       for (const std::string& cert : cert_files) {
-        std::error_code ec;
-        if (std::filesystem::exists(cert, ec)) {
+        if (file_exists(cert)) {
           cert_file_ = cert;
           return;
         }

